#151515982 enhance UI and implement feedback

AnthonyGW opened the pull request:
#### What does this PR do?
Improve on the UI made in development
Implement feedback received in previous PullRequests, Code Review and Code Defense.

#### Description of Task to be completed?
Use bootstrap to style the pages and forms.
Feedback Implemented:
1. Add comments to code.
2. Break long lines
3. Add newline to end of files
4. Remove unused imports

#### How should this be manually tested?
There are no test files implemented yet.
Test functionality by interacting with the GUI forms and buttons, the described routes have respective links to go to them:
1. On the base route (`/`), Click on the button `Register` to use create a new user by providing a username, email address and password in the form provided. App should open the Login form when successful.
2. Log in with account email address and password. App should redirect to user dashboard when successful.
3. Create a new bucketlist with the form to the left of the page.
4. View the bucketlists created to the right of the page.
5. Search for bucketlists by name using the search bar.
6. View the rest of the user's bucketlists (when they are more than 10) using the page navigation buttons.
7. Edit or delete a specific bucketlist using the links in the same row. User can leave the form fields blank to keep the same information when editing.
8. View a specific bucketlist's items by clicking on its name.
9. Create a new bucketlist item using the form to the left of the page.
10. View bucketlist items to the right of the page.

#### Any background context you want to provide?
The UI is based on the challenge 2 bootcamp project which can be found here: [BucketListUltima](https://github.com/AnthonyWaithaka/Bucket-List-Ultima)

Pivotal Tracker story: <https://www.pivotaltracker.com/story/show/151513228>

#### Screenshots
<img width="1280" alt="screen shot 2017-10-03 at 16 26 58" src="https://user-images.githubusercontent.com/31277260/31127540-f674188e-a857-11e7-8647-654683f3536e.png">

<img width="910" alt="screen shot 2017-10-03 at 16 27 07" src="https://user-images.githubusercontent.com/31277260/31127565-081ac3f8-a858-11e7-9f16-246d990fbb54.png">

<img width="858" alt="screen shot 2017-10-03 at 16 27 17" src="https://user-images.githubusercontent.com/31277260/31127592-191cda88-a858-11e7-95ce-44869ea208f4.png">

<img width="1256" alt="screen shot 2017-10-03 at 16 27 36" src="https://user-images.githubusercontent.com/31277260/31127619-2a018b50-a858-11e7-91ce-bc678a1d74a7.png">

<img width="1243" alt="screen shot 2017-10-03 at 16 28 22" src="https://user-images.githubusercontent.com/31277260/31127734-570aaaf0-a858-11e7-9cf9-37fb6195a353.png">

AnthonyGW committed:
feat(store): add store for user authentication

AnthonyGW committed:
feat(route): add bucketlists route

AnthonyGW committed:
feat(bucketlists): add page for bucketlists CRUD

AnthonyGW committed:
feat(dispatch): set flux dispatcher and actions

feat(dispatcher): initialize a dispatcher

feat(actions): define actions for the stores

fix(import and setStore): import actions and set store data

update(readme): add information about the project

fix(title): fix markdown format

AnthonyGW committed:
feat(all): bucketlists feature

feat(bucketlists): add button to view bucketlists

- Link to bucketlists page

feat(resetToken): function to handle login and logout changes

fix(all): remove authentication tokens from parameters

feat(bucketlist): add bucketlist module

- module can be repeatedly called and rendered

fix(all): render bucketlist components from store data

- store user's bucketlists in state
- update state when bucketlist store is updates
- reload bucketlists from API every time component is loaded
- delete bucketlist on button click

fix(all): delete unused token store

fix(response): store authentication token and reload bucketlists from API

fix(import): remove comment on import

fix(all): rename bucketlist store as js

- renamed from jsx
- updated to handle post, get and delete requests

AnthonyGW committed:
fix(readme): readme update and minor bug fix

update(readme): add instructions for demo and the current features

update(readme): add instructions for demo and the current features

fix(retrieve): add handling for 404 error

AnthonyGW committed:
feat(all): logout and auto navigation

feat(all): add logout button and functions to handle logout

fix(handleSubmit): navigate to bucketlists route on login

fix(handleSubmit): navigate to login route on creating account

AnthonyGW committed:
feat(bucketlistitems): add route for bucketlist items

- also rename some routes and links to begin with /auth/

AnthonyGW committed:
feat(all): render bucketlist items retrieved from api

feat(all): functions that dispatch commands for manipulating bucketlist items

AnthonyGW committed:
feat(store): use remote api

fix(all): use herokuapp instead of local server api

fix(catch_error): remove handling for 404 errors

update(readme): add new features to description

update(readme): add instruction for running the app

AnthonyGW committed:
feat(all): edit bucketlists feature

feat(import): import bootstrap css

feat(all): add edit bucketlist modal

- store form input in state
- handle form submit for both create and edit in the same method
- handle modal open and close through method and state

feat(editBucketlist): add action to edit bucketlist

feat(editBucketlist): handle edit bucketlist action

- order the retrieved bucketists by most recently updated
- send PUT request to server
- handle action case for edit bucketlist

fix(BucketlistStore): get data from key bucketlists

AnthonyGW committed:
feat(all): add edit item modal

- store form data in state
- initialize modal in state
- handle modal open and close in function
- submit edit item data through the same function as create item

feat(editItem): add edit item action

fix(editItem): rename type

- rename EDIT_ITEMS to EDIT_ITEM

feat(editItem): handle edit item action

- send PUT request for the item to be edited
- order the items retrieved by recently updated

AnthonyGW committed:
fix(logout): url fix

AnthonyGW committed:
feat(all): update homepage and dashboard UI

index.js
- remove links from index
- remove routes to register and login components

home.jsx
- render options to register and login users

sample.png, newlogo_lg.png
- use images in home page

bucketlists.js
- import bootstrap and router components
- pass query parameters for get request
- compute page numbers and items
- set active page and redirect state values
- implement logout, search, select active page and remaining days
- render returned bucketlist items in a table
- render buttons to navigate pages
- conditional redirect based on state value
- render navigation bar
- use selection list for date in place of text field
- arrange page in panels
- use bootstrap components for all forms

bucketlistactions.js
- add page tracking to delete function
- add search function

BucketlistStore.js
- store get request return value in state
- pass query parameters in get request
- pass page number after delete request
- add handling for search request

AnthonyGW committed:
feat(all): update bucketlist items UI

bucketlist_items.jsx
- import bootstrap components
- pass query parameters into get request function
- set state values for page numbers
- set state value for active page
- pass active page number to delete function
- add methods to handle search, result page navigation and logout
- show get results in table and paginate results
- handle redirect options from state values
- add navbar
- style form inputs with bootstrap

itemactions.js
- add active page parameter to delete action
- add search item action

AnthonyGW committed:
update(all): update for changes in api

- save values returned by api
- pass query parameters to get request method
- pass active page parameter to delete method
- handle search item action

AnthonyGW committed:
fix(all): implement feedback from pull request

- 15 files changed, 6 deleted

- Delete unused files; history.js, nav.jsx, login.jsx, register.jsx, bucketlist.jsx, item.jsx, nav.jsx

index.js
- add comments to document code
- remove unused method handleLogout and related import

authactions.js, bucketlistactions.js, itemactions.js
- add comments and end-of-file newline

bucketlist_items.jsx, bucketlists.jsx
- add comments
- remove deleted components Item/Bucketlist and their import
- break long lines
- add newline at end of file

BucketlistStore.js, ItemStore.js
- add comments
- add newline at end-of-file

AnthonyGW commented:
Remaining with validating form inputs to avoid double entries/ invalid entries and a toggle for the search bar in bucketlists view to search for either bucketlists or bucketlist items.

jonathankamau commented on src/Components/nav.jsx:
```diff
@@ -1,51 +0,0 @@
-            </nav>
-        )
-    }
-}
```
new line needed here

jonathankamau commented on src/actions/itemactions.js:
```diff
@@ -0,0 +1,56 @@
+// src/actions/itemactions.js
+// Actions defined here as functions can be called throughout the app to operate on the bucketlist
```
I see that the operations defined in this file have to do with the bucketlist items not bucketlist.

AnthonyGW replied:
The comment continues on the next line to mention the item store

jonathankamau commented on src/pages/bucketlist_items.jsx:
```diff
@@ -0,0 +1,292 @@
+
+    render(){
+        let bucketlistItems = this.state.bucketlist_items;
+        console.log("captured items: ", bucketlistItems);
```
why this `console.log`?

AnthonyGW replied:
It was to track the items that were saved in the store and passed to the component. Mostly for development purposes, it's not needed in the production build

AnthonyGW committed:
fix(all): apply feedback

- remove console messages
- fix date default
- add back button to bucketlist item page
- put home icon in homepage
